lsp/nls/src/world.rs

```diff
@@ -216,18 +237,43 @@ impl World {
 
     /// Returns all of the diagnostics encountered during parsing.
     pub fn parse(&mut self, file_id: FileId) -> Vec<SerializableDiagnostic> {
-        let mut errs = nickel_lang_core::error::ParseErrors::default();
-        self.analysis_reg
-            .insert_with(|init_term_env, init_type_ctxt| {
-                let analysis = PackedAnalysis::parsed(
-                    file_id,
-                    &self.sources,
-                    init_term_env.clone(),
-                    init_type_ctxt.clone(),
-                );
-                errs = analysis.parse_errors().clone();
-                analysis
-            });
+        let errs = match self.file_format(file_id).unwrap_or_default() {
```

Comment:
`file_format` is failing if we can't find the source id in the cache. I find it a bit strange to use the default value (`InputFormat::Nickel`) when that happens. Shouldn't we throw an error, panic, return empty diagnostics or something like that?

Reply:
Sure, I've turned this one into a panic (since we panic in a bunch of places on invalid file ids)